### DISCUSSION TITLE
inline python code not interpreting

### DISCUSSION THREAD

**Discussion by combiz** [quarto-dev/quarto-cli#3570] [Q&A] [answered]
I've used this pattern extensively with rmarkdown, but the output isn't as expected with quarto:

e.g.
"The data frame contains `python len(df)` rows."

This simply outputs the same line with code highlighted, instead of interpreting the python command. Any ideas?

**Comment by rgaiacs** [comment]
Thanks for the question @combiz .

As far as I know, [knitr](https://yihui.org/knitr/) and [rmarkdown](https://rmarkdown.rstudio.com/) only support [inline code](https://rmarkdown.rstudio.com/lesson-4.html) in the R language. 

Consider the follow minimal working `.Rmd` file

```
---
title: "Untitled"
author: "Raniere Silva"
date: "2022-12-06"
output: html_document
---

Foo `r 1 + 1`

Bar `python 1 + 1`
```

The output of `rmarkdown::render('mwe.Rmd', 'md_document')` is

```
Foo 2

Bar `python 1 + 1`
```

[As mentioned in StackOverflow](https://stackoverflow.com/a/69622352/1802726), you can use the reticulate package to have inline code access to Python variables.

**Comment by cscheid** [comment]
Yes, unfortunately that inline syntax isn't supported by Jupyter, and we at Quarto are waiting for a portable solution across engines.

In this case, you can use jupyter's computational Markdown cells instead, and output the entire paragraph at execution time:

````
```{python}
from IPython.display import display, Markdown
foo = 5
display(Markdown('A paragraph, and a variable `foo` with value `%s`.' % foo'))
```
````

(I haven't tested this one so I might have the syntax slightly wrong, but hopefully the gist is clear.)

**Comment by combiz** [comment]
Thank you both @rgaiacs @cscheid, these suggestions are helpful.

**Comment by cderv** [comment, marked as answer]
FWIW There has been a recent addition for this in 1.4 : https://github.com/quarto-dev/quarto-cli/pull/6190
